### ISSUE TITLE
chromebook_kukui: problem: kappa: High pitch noise - Debian 11

### ISSUE THREAD

**Issue by aneeshlingala** [hexdump0815/imagebuilder#65]
A high pitch noise comes sometimes when system wakes up from sleep. And it can only be stopped by turning off the system. Please fix this, and this is on the HP Chromebook 11a (kappa).

**Comment by hexdump0815** [comment]
i so far not have observed this on my kappa or anorther kukui device, but i have observed it on an intel celeron chromebook running linux as well - i think what helped there was to play some sound to stop that strange noise ... it might be worth a try to add playing a sound file with a few seconds of silence on wakeup to avoid something like this ... i'll have a look at this, but it will take a few days i guess ... can you meanwhile maybe verify if playing some sound helps to stop the noise? maybe just have a browser tab with a youtube video around which you start playing as soon as the noise appears again on wakeup.

**Comment by aneeshlingala** [comment]
Sure, could it be something related to the ucm file ported from chromeOS, or it could be alsa freaking out.

**Comment by hexdump0815** [comment]
i do not think that it is related to ucm, i guess its more a problem with the reinitialization of the audio hardware after resume (maybe some timing issue, so that it only happens from time to time)

**Comment by aneeshlingala** [comment]
Oh, still haven't got the high-pitched noise yet, I will make a comment when it happens.

**Comment by aneeshlingala** [comment]
Well, I got the High-Pitched noise and muting didn't help, I couldn't play a video in time as it stopped 6 seconds after.

**Comment by hexdump0815** [comment]
what did stop 6 seconds after?

some more ideas to debug the problem:
- what happens if you put the system back into sleep again immediately and wake it up a little later?
- to rule out pulesaudio try: "systemctl --user stop pulseaudio.service" as the logged in user and check if that stops the noise (i guess it will not)
- try to reload the alsa ucm config (run as root): "alsaucm -c mt8183_mt6358_t reload" or "alsaucm -c mt8183_mt6358_t reset"
- try to unload the sound driver (run as root): "rmmod mt8183_mt6358_ts3a227_max98357 snd_soc_mt8183_afe snd_soc_mtk_common snd_soc_ts3a227e snd_soc_mt6358 snd_soc_max98357a"

good luck and best wishes - hexdump

**Comment by aneeshlingala** [comment]
Sure, will try when im free

**Comment by aneeshlingala** [comment]
Sadly, i'm not using the device anymore, so I gave it to someone and reinstalled chromeOS on it, so I will close this issue.

**Comment by aneeshlingala** [comment]
I got the laptop back again.

**Comment by aneeshlingala** [comment]
Well, I played a video when this happened a few days ago, it did not stop the high pitched noise

**Comment by hexdump0815** [comment]
ok - thanks for the report - no real idea yet how to get this resolved - will maybe try to daily drive my kappa soon to see if i run into the same problem with it

**Comment by aneeshlingala** [comment]
Sure.

Thanks, Aneesh

**Comment by hexdump0815** [comment]
@aneeshlingala - just a little update: i used my kappa for a week now as my main daily system and was not able to reproduce the problem - no strange noise and i did use it with sound, without sound, put it into sleep mode a few times per day etc. ... this was with debian bullseye upgraded to bookworm with the normal xfce desktop

**Comment by hexdump0815** [comment]
ok - got it this morning myself as well while resuming from suspend - putting it back into suspend and resume once more fixed the problem (i.e. no more strange sound) ... it looks like the problem was there the following message showed up in dmesg:
```
mt8183-audio 11220000.audio-controller:mt8183-afe-pcm: mtk_afe_pcm_pointer hw_ptr err
```

**Comment by aneeshlingala** [comment]
Glad to hear it, I am currently using chromeOS in a need of stability, I
will reinstall soon.

Thanks,
Aneesh

**Comment by hexdump0815** [comment]
i have seen this on juniper now as well - so it sems to affect multiple kukui devices, maybe all with this audio setup or maybe even all kukui devices

**Comment by aneeshlingala** [comment]
On kernel 6.1.11, I have not experienced the noise. Closing the issue.

**Comment by hexdump0815** [comment]
yes also i did not yet notice it anymore for a long time since using the v6.1.11 kernel, so i can confirm your observation

**Comment by aneeshlingala** [comment]
Hello Hexdump0815,

Reopening the issue. Few days ago, the high pitched noise came back. This is on the kernel 6.1.11-stb-mt8+.

Thanks,
Aneesh

**Comment by hexdump0815** [comment]
oh - that is bad news then - but as the cause of the problem is still unknown there is no other option than the workaround to put system to sleep and wake it up again ... lets see, hopefully some kernel update one day will fix it

i'm using my kappa as daily driver again right now using the v6.1.11 kernel and did not run into the problem again yet.

**Comment by aneeshlingala** [comment]
Good news:
It happened again, but I have noticed it only happens for a few seconds instead.

**Comment by willmasters** [comment]
I also get the crazy long beep after resuming on krane in the September release, it's very loud and distressing. X)

Can't really take it out the house because of it. People will probably thing there is a fire alarm going off. :)

**Comment by hexdump0815** [comment]
@willmasters - i so far only got it on kappa and never on krane yet and that the suspend/resume workaround does not help on your device does not sound very good ... sadly i still have no idea how to get rid of it

**Comment by willmasters** [comment]
Thanks @hexdump0815 . :) And no worries, if I find out anything further - I will post it here.

**Comment by aneeshlingala** [comment]
One extra thing:

This happens on Postmarketos and my custom Arch Linux Image.

**Comment by hexdump0815** [comment]
just a little update: i'm running currently some v6.6 kernel on my kappa and i haven't heard the bad sound for a long time - i'm not sure if its really gone or if i'm just lucky, but maybe newer kernels will bring some help with this - lets see what time will tell :)

**Comment by aneeshlingala** [comment]
Lets hope it has gone over time... And also, happy new year

**Comment by hexdump0815** [comment]
happy new year as well

it looks like the problem maybe manifests in a different way with newer kernels which is less annoying: i noticed that sometimes something seems to be wrong with the audio clocks resulting in audio or videos with audio are being played extremely slow - the "fix" to make it work is the same es in the past: just put the device into sleep and wake it back up and things are fine again ... so maybe some of the clocks are not restored properly on suspend/resume from time to time ...

**Comment by aneeshlingala** [comment]
i have also tested the new 6.6 kernel, and I haven't gotten any high pitch sound. I observed this has been fixed from kernel ver. 6.5.5. closing the issue

**Comment by hexdump0815** [comment]
today i got the noise once even with my v6.6.9 kernel, but that was after months of intensive use of my kappa chromebook as daily driver, so it looks like its not completely gone, but very very rare meanwhile ...

**Comment by aneeshlingala** [comment]
Weird, have been daily drive ring my kappa with KDE and there has been no
sound, also tried MATE, no high pitched sound as well. This might be
related to DE but I'm not too sure because it also happens on postmarketOS
with phosh

**Comment by aneeshlingala** [comment]
This has happened to me again on 6.12.20 cbm kernel, I will attach a video.

**Comment by aneeshlingala** [comment]
I wasn't able to upload a file on GitHub, I have uploaded the video on a separate site. [https://jumpshare.com/embed/QaBZ3VWExICFTH30rYax](https://jumpshare.com/embed/QaBZ3VWExICFTH30rYax)

**Comment by aneeshlingala** [comment]
Again, the logs show `mt8183-audio 11220000.audio-controller:mt8183-afe-pcm: mtk_afe_pcm_pointer hw_ptr err ` being spammed in dmesg

**Comment by thenameisluk** [comment]
yeah we still don't know the cause of this issue

but if you run (when it happens)
```
systemctl restart --user pipewire
```
it should get rid of the noise (for the time being)

sad thing is we can't even git bisect this since there was no version without this issue afaik

**Comment by aneeshlingala** [comment]
Issue has been on and off for 3 years. It was fine until 6.6 kernel. Most
likely cbm merge did something?

**Comment by aneeshlingala** [comment]
However, this only happened once so we still need to see if it is a rare occurrence like 6.6

**Comment by hexdump0815** [comment]
@aneeshlingala - i guess it was always there, it just happens so rarely and randomly that sometimes one might think its gone - and then it comes back one day :) ... for me i have not run into it for a long time (using my kappa as daily driver), but from time to time the audio clocks seem to be off (i.e. videos play back extremely slow) and i have to suspend/unsuspend to make it work again ... will have to give the pipewire/pulseaudio restart a try ...

**Comment by hexdump0815** [comment]
just to have it as a reference here - this one might be related: https://patchwork.kernel.org/project/linux-mediatek/cover/20250424102509.1083185-1-wenst@chromium.org/ ... in the recent times (=months) i got the high pitch sound only once or twice but a few more times situations where the sound was either extremely slow (another suspend/resume usually fixed that) or not working at all (reboot required to get the sound back) ... in the non working case i noticed some memory allocation error in the sound driver in the logs, so it might be related to that patch series which somewhere also mentions memory starvation ... as i read it this patch series is still wip and not yet ready completely for mt8183, but definitely something to keep an eye on over the next months ...

**Comment by hexdump0815** [comment]
it also happens with pulseaudio, it can just take quite a while until it happens

**Comment by thenameisluk** [comment]
@hexdump0815 given that the noise only occures after suspense could this fix it?

at ```/etc/systemd/system/fix-pipewire.service```
```
[Unit]
Description=stop high pitch
After=suspend.target hibernate.target hybrid-sleep.target suspend-then-hibernate.target


[Service]
ExecStart=systemctl restart --user -M linux@ pipewire

[Install]
WantedBy=suspend.target hibernate.target hybrid-sleep.target suspend-then-hibernate.target
```

the ```systemctl restart --user pipewire``` is already known to get rid of the noise when it occures
(and the ```-M linux@``` part if to do it for specyfic user since this one is ran by root)

it is a hack but unless someone finds a solution it might be better than nothing

**Comment by hexdump0815** [comment]
@thenameisluk - i'm still using pulseaudio and there restarting oes not seem to help (tried it in a few cases), but for pipewire we should at least document what you sugested as a workaround (or point to your last post in the kukui specific doc maybe)

**Comment by MuRongBAD** [comment]
> [@hexdump0815](https://github.com/hexdump0815) given that the noise only occurs after suspense could this fix it?
> 
> at `/etc/systemd/system/fix-pipewire@.service`
> 
> [Unit]
> Description=stop high pitch
> After=suspend.target hibernate.target hybrid-sleep.target suspend-then-hibernate.target
> 
> 
> [Service]
> ExecStart="systemctl restart --user -M %I@ pipewire"
> 
> [Install]
> WantedBy=suspend.target hibernate.target hybrid-sleep.target suspend-then-hibernate.target
> then
> 
> systemctl enable fix-pipewire@linux
> the `systemctl restart --user pipewire` is already known to get rid of the noise when it occurs (and the `-M linux@` part is to do it for specific user since this one is ran by root)
> 
> it is a hack but unless someone finds a solution it might be better than nothing

Unfortunately I found the problem appeared on my device (Lenovo chromebook duet(google-krane)).The high pitch sound appears multiple times a day(randomly?) when I tries to wake up my device from sleeping.
Using 6.12.36,postmarketos,gnome,pipewire.
I'll try to use this method.

**Comment by MuRongBAD** [comment]
Update:doesn't work.

**Comment by mikobuntu** [comment]
Hi guys, some systems might be also be using pipewire-pulse , so maybe change the line in the [Service] section to reflect that.
 If/when I figure out an easy method to include more than 1 command in the 'ExecStart' then both 'pipewire' and 'pipewire-pulse' services could be restarted ( so we can possibly not get the beep when shutting the lid and resuming while actually playing a video/Audio etc ) .

**Comment by hexdump0815** [comment]
just to have this case documented as well: today i had a very annoying instance of this problem - after resume i got the high pitch noise and putting my kukui kappa (with v6.12.42 kernel) into suspend did not help - after resuming the noise was back each time (i tried it many times) and i had to reboot to get rid of it ... this is the first time for me that that did not help

**Comment by aneeshlingala** [comment]
ok

**Comment by hexdump0815** [comment]
inspired by https://github.com/velvet-os/imagebuilder/issues/388#issuecomment-3394082946 i'm asking myself if anyone running gnome or kde on its kukui device ever had this high pitch noise problem? for people who have this high pitch noise problem in xfce and debian trixie it might be worth a try to switch to pipewire via "apt-get install pipewire-audio" which should be a transparent dropin for the default pulseaudio and see if the problem still comes up or not. i think the chance is not that high that this might have any effect, but it could be and is definitely worth a try i would say.

**Comment by aneeshlingala** [comment]
I had it in KDE 5 and 6

**Comment by hexdump0815** [comment]
@aneeshlingala - thanks for this info - so pipewire will not help for this problem (as i guessed)

**Comment by hexdump0815** [comment]
it looks like it happens more often now that the ugly sound does not go away via some new supsend/resume cycle - i now had it three times this way with v6.12.32 and v6.12.50 kernels, so maybe its more the case with newer kernels.

i have found a workaround to at least get rid of the beep when it happens - as root run:
```
amixer sset 'I2S3_CH1 DL1_CH1' off
amixer sset 'I2S3_CH2 DL1_CH2' off
```
that takes the beep sound off for me - but to get audio back working, a reboot is still required in such cases ... at least this is some working last resort to avoid having to reboot immediately due to that sound ...

**Comment by hexdump0815** [comment]
i meanwhile did a "apt-get install pipewire-audio" which transparently replaces pulseaudio with pipewire on my daily driven kappa and with that it works as mentioned by @thenameisluk here: https://github.com/velvet-os/imagebuilder/issues/65#issuecomment-2795854869 that i can get rid of the beeping sound via "systemctl restart --user pipewire" as logged in user and sound is still working afterwards - so far the best workaround for the problem it seems ...

**Comment by hexdump0815** [comment]
is it just me or is the problem gone? maybe i should not say it too loud :) ... on my daily used kappa which is regularly updated to latest debian trixie updates and running with pipewire and the 6.12.50 cbm kernel i did not run into this problem for more than the last three months now ... did anyone else have the same experience or otherwise? maybe some debian updates fixed something triggering the problem?

**Comment by hexdump0815** [comment]
update: sadly i had the high pitch sound again today - so looks like the problem is still there ...

update 2: funny - after not seeing this problems for more than 3 months i now saw it multiple times in a row :)

**Comment by hexdump0815** [comment]
as i seem to get this problem now quite often again, i played around with it a bit more and i think i have maybe an idea of how to work around it. i'm meanwhile using pipewire on my kappa ("apt-get install pipewire-audio" will switch the default xfce pulseaudio install to pipewire) and with pipewire it seems to work to just stop and restart pipewire when the strange sound appears to get rid of it (at least most of the time). also the strange sound only appeared when resuming from suspend. my idea was now to simply stop pipewire before suspend and start it again after resume and i tried this manually a few times via "systemctl --user stop pipewire ; systemctl --user stop pipewire.socket" before suspend und "systemctl --user start pipewire ; systemctl --user start pipewire.socket" after resume and it looks like this might help to avoid the strange sound to appear (still more testing required).

what would be required now is to integrate those steps into the systemd suspend/resume handling - one thing which is a bit special about it is that it has to be run in the user session. so if someone has a bit of time to look into if it is possible to cleanly integrate this into the systemd suspend/resume handling it would be very good.

**Comment by archisman-panigrahi** [comment]
The high pitch sound occurs in some Intel Chromebooks as well. There, a permanent solution exists to turn it off once and for all: https://askubuntu.com/q/1451125/124466. While this solution was Intel specific, maybe someone can use this to figure out how to apply it for ARM Chromebooks?

**Comment by hexdump0815** [comment]
@archisman-panigrahi - that does not apply to the problems we see with kukui devices - that are two different problems ... i actually had that intel problem on some intel chromebook as well some years ago :)

**Comment by aneeshlingala** [comment]
https://github.com/FyraLabs/submarine

I will try this with an arch linux iso, and see if the issue reproduces (if sound even works!)

**Comment by hexdump0815** [comment]
with v6.18 audio was not working at all anymore on kukui and i always got the error mentioned here: https://github.com/velvet-os/imagebuilder/issues/65#issuecomment-1295751273 and with the v6.18.30 test kernel here: https://github.com/velvet-os/imagebuilder-testkernels/releases/tag/spring-test-kernels i synced with latest postmarketos mediatek kernel patches which also brought some uncommented power-domain fix which fixes audio for v6.18, gets rid of the above mentioned error message and as a result maybe also fixes this high pitch problem ... at least there is some hope - in case anyone still sees that problem with that or a newer kernel, please let me know here

**Comment by hexdump0815** [comment]
hope is over - i had the noise again with a self built v6.18.33 kernel :(
